src/libraries/Native/Unix/System.Security.Cryptography.Native/extra_libs.cmake

```diff
@@ -13,7 +13,7 @@ macro(append_extra_cryptography_libs NativeLibsExtra)
     
     if (FEATURE_DISTRO_AGNOSTIC_SSL OR CLR_CMAKE_TARGET_OSX OR CLR_CMAKE_TARGET_MACCATALYST)
         # Link with libdl.so to get the dlopen / dlsym / dlclose
-        list(APPEND ${NativeLibsExtra} dl)
+        list(APPEND ${NativeLibsExtra} dl pthread)
```

Comment:
This is not fully correct for adding pthread support. Instead of the pthread library, the compiler `-pthread` option should be used. See https://stackoverflow.com/questions/23250863/difference-between-pthread-and-lpthread-while-compiling
The option should be added to the CMakeLists.txt in the same directory as this extra_libs.cmake file using `add_compile_options(-pthread)` somewhere after the `FEATURE_DISTRO_AGNOSTIC_SSL` is set (so that the same cmake condition as here can be used to wrap the `add_compile_options`.

Reply:
Hm.  I tried

```diff
--- a/src/libraries/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt
+++ b/src/libraries/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt
@@ -82,6 +82,8 @@ set_target_properties(System.Security.Cryptography.Native.OpenSsl-Static PROPERT

 if (GEN_SHARED_LIB)
     if (FEATURE_DISTRO_AGNOSTIC_SSL)
+        target_compile_options(System.Security.Cryptography.Native.OpenSsl PRIVATE "-pthread")
+
         # on macOS the link step fails with undefined symbols, and the script doesn't run.
         # if the build succeeds, the script would succeed, except it uses a Linux-only command.
         #
diff --git a/src/libraries/Native/Unix/System.Security.Cryptography.Native/extra_libs.cmake b/src/libraries/Native/Unix/System.Security.Cryptography.Native/extra_libs.cmake
index 84a5b89a4c5..422b2b04cb1 100644
--- a/src/libraries/Native/Unix/System.Security.Cryptography.Native/extra_libs.cmake
+++ b/src/libraries/Native/Unix/System.Security.Cryptography.Native/extra_libs.cmake
@@ -13,7 +13,7 @@ macro(append_extra_cryptography_libs NativeLibsExtra)

     if (FEATURE_DISTRO_AGNOSTIC_SSL OR CLR_CMAKE_TARGET_OSX OR CLR_CMAKE_TARGET_MACCATALYST)
         # Link with libdl.so to get the dlopen / dlsym / dlclose
-        list(APPEND ${NativeLibsExtra} dl pthread)
+        list(APPEND ${NativeLibsExtra} dl)
     else()
         list(APPEND ${NativeLibsExtra} ${OPENSSL_CRYPTO_LIBRARY} ${OPENSSL_SSL_LIBRARY})
     endif()
```

With all of PRIVATE, PUBLIC, and INTERFACE.  None worked.  (Posting this update because I have to run to an appointment)

Reply:
Ah, you need to add it both via add_compile_options and add_link_options. You don't need to use the target_compile_option, as the current target is the implicit one to which the two previously mentioned functions add the options.

Reply:
Hm, didn't have success with add_link_options/add_compile_options, but did with target_link_options/target_compile_options.

Reply:
(The make step worked, bit it still produced an "Undefined symbol(s) pthread_once" error)

Reply:
That's strange, it worked fine for me locally the way I've suggested.

Reply:
Ah, you've put it after the add_library. My way need it before that. Basically, the preferred way is to set options / defines and then add_library or add_executable. That's what we use in the whole runtime except for one or two places where the other way is needed.

Reply:
Ah.  I think the fact that I landed on target_compile_options first required that a target be defined, so I ended up in the bottom.  But now it should be happy.